### PR TITLE
Improve README.md, mention license, rename a file

### LINKED ISSUE #1273: Clarify GAP licensing terms: GPL 2 or later; in particular, GPL 3 is OK
GAP has been licensed under the "GPL 2 or later" for a long time. Indeed, `doc/ref/copyrigh.xml`  says so. (Off-topic: can we rename that file to `copyright.xml`, please?)

But this is not so clear when looking at other places. For starters, I think our `README.md` should explicitly state the license (using similar verbiage as in the manual), and state some copyright.

We might also want to add this in other places, not sure where, though.

BTW, I just noticed we have text like "Copyright © (1987-this year)" -- I am pretty sure that's not legally valid *shrug*.

### PR DESCRIPTION
**Pull request by fingolfin**
Resolves #1273

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # What is GAP?
 
 GAP is a system for computational discrete algebra, with particular emphasis on
-Computational Group Theory. GAP provides a programming language, a library of
+computational group theory. GAP provides a programming language, a library of
```
Comment (alex-konovalov) [inline README.md]: Agree - perhaps same should be done for the website and maybe manual, for consistency? (not necessarily in this PR)

File: README.md [hunk 2]
```diff
@@ -14,8 +14,6 @@ rings, vector spaces, algebras, combinatorial structures, and more.
 The system, including source, is distributed freely. You can study and easily
 modify or extend it for your special use.
 
-In July 2008, GAP was awarded the ACM/SIGSAM Richard Dimick Jenks Memorial Prize
```
Comment (alex-konovalov) [inline README.md]: Fine with me to omit this from README

File: README.md [hunk 3]
```diff
@@ -24,45 +22,49 @@ The latest stable release of the GAP system together with all currently redistri
...
-* GNU autoconf >= 2.69
+Alternatively, you can compile the latest development version of GAP. However, most
+users should instead use the latest official release instead.
+
+If you still wish to continue, you can obtain the latest GAP code from its git
+repository at
+  <https://github.com/gap-system/gap>
+In order to compile GAP, at least the following dependencies need to be installed:
+
+* a C compiler, e.g. GCC or clang
```
Comment (alex-konovalov) [inline README.md]: clang -> Clang

File: README.md [hunk 4]
```diff
@@ -24,45 +22,49 @@ The latest stable release of the GAP system together with all currently redistri
...
+* GNU autoconf (we recommend 2.69 or later)
+* GNU make
```
Comment (alex-konovalov) [inline README.md]: Hm, actually official names here are GNU Autoconf, GNU Make. But then I am imagining users typing Autoconf/Make/Clang etc. and getting errors. Maybe then GCC -> gcc for consistency?

File: doc/gapmacrodoc.tex [hunk 5]
```diff
@@ -26,7 +26,7 @@
 }
 \OneColumnTableOfContents
 \FrontMatter
-%\input ../doc/ref/copyrigh
+%\input ../doc/ref/copyright
```
Comment (alex-konovalov) [inline doc/gapmacrodoc.tex]: I think this line was commented out already, and may simply be removed.

File: README.md (hunk 1 above)
Comment (fingolfin) [reply README.md]: I don't see this appear in the manual, and obviously I can't fix it on the website with this PR.

File: README.md (hunk 4 above)
Comment (fingolfin) [reply README.md]: I'd not overthink this. People need to figure package names anyway (assuming they are on a Unix with a package manager; Windows users have a problem anyway). I suggested elsewhere that we could somewhere list typical package names for some major systems. E.g.:

>  On Ubuntu or Debian, you can install all packages required by GAP via
 > `sudo apt install build-essential autoconf automake libtool libreadline-dev libgmp-dev`
> On Fedora, use ...
> On DragonFlyBSD, use
> On macOS, use ...
>  *  ... if you are using homebrew: `brew install ...`
>  * ... if you are using Fink: `fink install 
>  * ... if you are using MacPorts: `

Etc. However, I am afraid this list might get a bit longish, and it interrupts the reading flow. Hmm.. In general, a section for macOS, and another for Windows, might be helpful. I am not sure the `README.md` is the place for all this... indeed, perhaps I should remove mention of the compiler etc. again, and instead point people to `INSTALL.md` ?

[also, the list I added in the README is missing libtool]

**Comment by alex-konovalov** [comment]
@fingolfin thanks, new version of this PR looks good to me.

**Pull request merged**